### PR TITLE
Workbox doc tidy

### PR DESCRIPTION
**Pull request by gauntface**
- Tidies the code for reference doc generation into a set of utility functions
- Moves workbox specific gulp tasks into a directory
- Builds the latest docs for workbox 2.1.2
- Builds the latest docs for workbox 3.0.0-alpha.1
- Fixes bug for contributors URL being for API instead of HTML URL
- Adds ability to reference the latest CDN URL instead of a hardcoded URL.

### CODE REVIEW

File: gulp-tasks/reference-docs/filter-tags-to-build.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/reference-docs/filter-tags-to-build.js]: Please add a JSDoc description to this file, include at least `@fileoverview` and `@author`

File: gulp-tasks/reference-docs/get-latest-tags.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/reference-docs/get-latest-tags.js]: Please add a JSDoc description to this file, include at least `@fileoverview` and `@author`

File: gulp-tasks/reference-docs/build-js-docs.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/reference-docs/build-js-docs.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: gulp-tasks/reference-docs/build-js-docs.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+const gutil = require('gulp-util');
+const fs = require('fs-extra');
+const path = require('path');
+const glob = require('glob');
+const wfHelper = require('../wfHelper');
+
+const canFindJSDocConf = (jsdocConfPath) => {
```
Comment (petele) [inline gulp-tasks/reference-docs/build-js-docs.js]: Please document functions with JSDoc. Include at least a brief description, and `@param`/`@return` as appropriate.  Will still merge if not added.

File: gulp-tasks/reference-docs/build-reference-docs.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/reference-docs/build-reference-docs.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: gulp-tasks/reference-docs/build-reference-docs.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+const path = require('path');
+const os = require('os');
+const fs = require('fs-extra');
+
+const getLatestTags = require('./get-latest-tags');
+const filterTagsToBuild = require('./filter-tags-to-build');
+const getSourceCode = require('./get-source-code');
+const buildJSDocs = require('./build-js-docs');
+
+const buildReferenceDocs = (gitUrl, docPath, jsdocConfPath) => {
```
Comment (petele) [inline gulp-tasks/reference-docs/build-reference-docs.js]: Please document functions with JSDoc. Include at least a brief description, and `@param`/`@return` as appropriate.  Will still merge if not added.

File: gulp-tasks/reference-docs/filter-tags-to-build.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+const fs = require('fs-extra');
+const path = require('path');
+
+const filterTagsToBuild = (tags, docPath) => {
```
Comment (petele) [inline gulp-tasks/reference-docs/filter-tags-to-build.js]: Please document functions with JSDoc. Include at least a brief description, and `@param`/`@return` as appropriate.  Will still merge if not added.

File: gulp-tasks/reference-docs/get-source-code.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/reference-docs/get-source-code.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: gulp-tasks/reference-docs/get-source-code.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+const fs = require('fs-extra');
+const wfHelper = require('../wfHelper');
+
+const getSourceCode = (gitUrl, tag, tmpPath) => {
```
Comment (petele) [inline gulp-tasks/reference-docs/get-source-code.js]: Please document functions with JSDoc. Include at least a brief description, and `@param`/`@return` as appropriate.  Will still merge if not added.

File: gulp-tasks/wfHelper.js [hunk 10]
```diff
@@ -49,11 +52,11 @@ function promisedExec(cmd, cwd) {
       stdOut = stdOut.trim();
       stdErr = stdErr.trim();
       if (err) {
-        gutil.log(' ', cmdLog, chalk.red('FAILED'));
```
Comment (petele) [inline gulp-tasks/wfHelper.js]: Why remove this? If several commands are running asynchronously, you won't know which one failed.

File: gulp-tasks/wfHelper.js [hunk 11]
```diff
@@ -49,11 +52,11 @@ function promisedExec(cmd, cwd) {
...
+        gutil.log('    (CMD):', chalk.red('FAILED'));
         reject(err);
         return;
       }
-      gutil.log(' ', cmdLog, chalk.green('OK'));
```
Comment (petele) [inline gulp-tasks/wfHelper.js]: Why remove this? If several commands are running asynchronously, you won't know which one failed.

File: gulp-tasks/workbox/workbox-generate-cdn-include-v3.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/workbox/workbox-generate-cdn-include-v3.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: gulp-tasks/workbox/workbox-generate-contributors.js [hunk 13]
```diff
@@ -1,9 +1,11 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/workbox/workbox-generate-contributors.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: gulp-tasks/workbox/workbox-generate-contributors.js [hunk 14]
```diff
@@ -14,7 +16,7 @@ const MAX_COLLABORATORS = 20;
  */
 
 gulp.task(`workbox-generate-contributors`, function() {
-  var token = process.env.GITHUB_TOKEN;
+  let token = process.env.GITHUB_TOKEN;
```
Comment (petele) [inline gulp-tasks/workbox/workbox-generate-contributors.js]: we already have a `GIT_TOKEN`, how is this one different?

File: gulp-tasks/workbox/workbox-reference-docs-v3.js [hunk 15]
```diff
@@ -0,0 +1,57 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/workbox/workbox-reference-docs-v3.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: gulp-tasks/workbox/workbox-reference-docs.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
+'use strict';
```
Comment (petele) [inline gulp-tasks/workbox/workbox-reference-docs.js]: Add the JSDoc description to this file and include at least `@fileoverview` and `@author`

File: src/content/en/tools/workbox/_shared/contributors.html [hunk 17]
```diff
@@ -0,0 +1,181 @@
+{% framebox height="auto" %}
```
Comment (petele) [inline src/content/en/tools/workbox/_shared/contributors.html]: Why is it in a framebox? I don't see any JS. You could do it simply as an html or md include.

File: src/content/en/tools/workbox/_shared/workbox-sw-cdn-url.html [hunk 18]
```diff
@@ -0,0 +1 @@
+https://storage.googleapis.com/workbox-cdn/releases/3.0.0-alpha.1/workbox-sw.js
```
Comment (petele) [inline src/content/en/tools/workbox/_shared/workbox-sw-cdn-url.html]: what's this for?

Comment (gauntface) [reply src/content/en/tools/workbox/_shared/workbox-sw-cdn-url.html]: Its so the docs can tell the user to go `importScripts('https://storage.....');` and start using Workbox.

The gulp task generates this and it's included in any page that needs it.

File: src/content/en/tools/workbox/_shared/contributors.html (hunk 17 above)
Comment (gauntface) [reply src/content/en/tools/workbox/_shared/contributors.html]: Will remove but I think I added this so `_index.yaml` would be happy.

Comment (petele) [reply src/content/en/tools/workbox/_shared/contributors.html]: OH! Fudge. Imports into `_index.yaml` are a PITA! Uh, let's remove it for now, but we may need to be extra careful on this one. :/

File: src/content/en/tools/workbox/_shared/workbox-sw-cdn-url.html (hunk 18 above)
Comment (petele) [reply src/content/en/tools/workbox/_shared/workbox-sw-cdn-url.html]: OK - cool, thanks!

File: gulp-tasks/wfHelper.js (hunk 10 above)
Comment (gauntface) [reply gulp-tasks/wfHelper.js]: Removed because it balloons logs - added back in.

File: gulp-tasks/wfHelper.js (hunk 11 above)
Comment (gauntface) [reply gulp-tasks/wfHelper.js]: Added back in.

File: gulp-tasks/workbox/workbox-generate-cdn-include-v3.js (hunk 12 above)
Comment (gauntface) [reply gulp-tasks/workbox/workbox-generate-cdn-include-v3.js]: Done

File: gulp-tasks/workbox/workbox-generate-contributors.js (hunk 13 above)
Comment (gauntface) [reply gulp-tasks/workbox/workbox-generate-contributors.js]: Done.

File: gulp-tasks/workbox/workbox-generate-contributors.js (hunk 14 above)
Comment (gauntface) [reply gulp-tasks/workbox/workbox-generate-contributors.js]: This wasn't caught on first review of this code after you request the text file change. Fixed now.

File: gulp-tasks/workbox/workbox-reference-docs-v3.js (hunk 15 above)
Comment (gauntface) [reply gulp-tasks/workbox/workbox-reference-docs-v3.js]: Done.

File: gulp-tasks/workbox/workbox-reference-docs.js (hunk 16 above)
Comment (gauntface) [reply gulp-tasks/workbox/workbox-reference-docs.js]: Done.

File: gulp-tasks/reference-docs/get-source-code.js (hunk 8 above)
Comment (gauntface) [reply gulp-tasks/reference-docs/get-source-code.js]: Done.

File: gulp-tasks/reference-docs/get-latest-tags.js (hunk 2 above)
Comment (gauntface) [reply gulp-tasks/reference-docs/get-latest-tags.js]: Done.

File: gulp-tasks/reference-docs/filter-tags-to-build.js (hunk 1 above)
Comment (gauntface) [reply gulp-tasks/reference-docs/filter-tags-to-build.js]: Done.

File: gulp-tasks/reference-docs/build-reference-docs.js (hunk 5 above)
Comment (gauntface) [reply gulp-tasks/reference-docs/build-reference-docs.js]: Done.

File: gulp-tasks/reference-docs/build-js-docs.js (hunk 3 above)
Comment (gauntface) [reply gulp-tasks/reference-docs/build-js-docs.js]: Done.

File: gulp-tasks/wfHelper.js [hunk 19]
```diff
@@ -49,11 +52,11 @@ function promisedExec(cmd, cwd) {
       stdOut = stdOut.trim();
       stdErr = stdErr.trim();
       if (err) {
-        gutil.log(' ', cmdLog, chalk.red('FAILED'));
+        gutil.log('    (CMD):', cmdLog, chalk.red('FAILED'));
         reject(err);
         return;
       }
-      gutil.log(' ', cmdLog, chalk.green('OK'));
+      gutil.log('    (CMD):', cmdLog, chalk.green('OK'));
```
Comment (petele) [inline gulp-tasks/wfHelper.js]: Same comment here - would still like to see the cmd line executed

File: gulp-tasks/workbox/workbox-generate-contributors.js (hunk 14 above)
Comment (petele) [reply gulp-tasks/workbox/workbox-generate-contributors.js]: Yah, I realized that afterwards. It's the Git token for @WebFundBot. We should prob rename them at some point.

**Pull request merged**